doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+with the same length in equals signs.
+
+::
+
+  Explanation
+  ===========
+
+This section consists of a more elaborate description of what the function,
+class, or method does when the concise Single-Sentence Summary will not
+suffice. This section should be used to clarify functionality in several
+sentences or paragraphs.
+
+3. Examples Section
+-------------------
+
+This section is **required** for every docstring and checks will not pass if it
+is not included (WIP issue `#17643
+<https://github.com/sympy/sympy/issues/17643>`_). It should be labeled with the
+heading “Examples” (even if there is only one example) underlined with the same
+length in equals signs.
+
+::
+
+  Examples
+  ========
+
+This section consists of examples that show how the function works, called
+doctests. Doctests should be complicated enough that they fully demonstrate the
+API and functionality of the function, but simple enough that a user can
+understand them without too much thought. The perfect doctest tells the user
+exactly what they need to know about the function without reading any other
+part of the docstring.
+
+When multiple examples are provided, they should be separated by blank lines.
+Comments explaining the examples should have blank lines both above and below
+them.
+
+Do not think of doctests as tests. Think of them as examples that happen to be
+tested. They should demonstrate the API of the function to the user (i.e., what
+the input parameters look like, what the output looks like, and what it does).
+If you only want to test something, add a test to the relevant ``test_*.py file``.
+
+You can use the ``./bin/coverage_doctest.py`` script to test the doctest
+coverage of a file or module. Run the doctests with ``./bin/doctest``.
+
+You should only skip the testing of an example if it is impossible to test it.
+If necessary, testing of an example can be skipped by adding a special comment.
+
+**Example**
+
+>>> import random
+>>> random.random()      # doctest: +SKIP
+0.6868680200532414
+
+If an example is longer than 80 characters, it should be line wrapped. Examples
+should be line wrapped so that they are still valid Python code, using ``...``
+continuation as in a Python prompt. For example, from the ODE module
+documentation:
+
+**Example**
+
+>>> from sympy import Function, dsolve, cos, sin
+>>> from sympy.abc import x
+>>> f = Function('f')
+>>> dsolve(cos(f(x)) - (x*sin(f(x)) - f(x)**2)*f(x).diff(x),
+... f(x), hint='1st_exact')
+Eq(x*cos(f(x)) + f(x)**3/3, C1)
+
+Here ``dsolve(cos(f(x)) - (x*sin(f(x)) - f(x)**2)*f(x).diff(x), f(x), hint='1st_exact')`` is too long, so we line break it after a comma so that it
+is readable, and put ``...`` on the continuation lines. If this is not done
+correctly, the doctests will fail.
+
+The output of a command can also be line wrapped. No ``...`` should be used in
+this case. The doctester automatically accepts output that is line wrapped.
+
+**Example**
+
+>>> list(range(30))
```

Comment:
The example docstrings here are rendering as docstrings in the HTML. Maybe it would be better for the guide if they are listed as plain code blocks, like

```
.. code:: 

   >>> list(range(30))
   [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21,
   22, 23, 24, 25, 26, 27, 28, 29]
```

What do you think?

Reply:
But those code blocks wouldn't be doctested, or would they?

Reply:
I tried formatting these examples like you've shown, but the SymPy Live is still rendering. Are we trying to get rid of it? Can discuss on our call.